src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectGCDumpActionTests.cs

```diff
@@ -0,0 +1,102 @@
+        private readonly EndpointUtilities _endpointUtilities;
+
+        public CollectGCDumpActionTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+            _endpointUtilities = new(_outputHelper);
+        }
+
+        [Fact]
+        public async Task CollectGCDumpAction_Success()
+        {
+            using TemporaryDirectory tempDirectory = new(_outputHelper);
+
+            await TestHostHelper.CreateCollectionRulesHost(_outputHelper, rootOptions =>
+            {
+                rootOptions.AddFileSystemEgress(ExpectedEgressProvider, tempDirectory.FullName);
+
+                rootOptions.CreateCollectionRule(DefaultRuleName)
+                    .AddCollectGCDumpAction(ExpectedEgressProvider)
+                    .SetStartupTrigger();
+            }, async host =>
+            {
+                IOptionsMonitor<CollectionRuleOptions> ruleOptionsMonitor = host.Services.GetService<IOptionsMonitor<CollectionRuleOptions>>();
+                CollectGCDumpOptions options = (CollectGCDumpOptions)ruleOptionsMonitor.Get(DefaultRuleName).Actions[0].Settings;
+
+                ICollectionRuleActionProxy action;
+                Assert.True(host.Services.GetService<ICollectionRuleActionOperations>().TryCreateAction(KnownCollectionRuleActions.CollectGCDump, out action));
+
+                EndpointInfoSourceCallback callback = new(_outputHelper);
+                await using var source = _endpointUtilities.CreateServerSource(out string transportName, callback);
+                source.Start();
+
+                AppRunner runner = _endpointUtilities.CreateAppRunner(transportName, TargetFrameworkMoniker.Net60); // Arbitrarily chose Net60
+
+                Task<IEndpointInfo> newEndpointInfoTask = callback.WaitForNewEndpointInfoAsync(runner, CommonTestTimeouts.StartProcess);
+
+                await runner.ExecuteAsync(async () =>
+                {
+                    IEndpointInfo endpointInfo = await newEndpointInfoTask;
+
+                    using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.GCDumpTimeout);
+                    CollectionRuleActionResult result = await action.ExecuteAsync(options, endpointInfo, cancellationTokenSource.Token);
+
+                    Assert.NotNull(result.OutputValues);
+                    Assert.True(result.OutputValues.TryGetValue(CollectionRuleActionConstants.EgressPathOutputValueName, out string egressPath));
+                    Assert.True(File.Exists(egressPath));
+
+                    using FileStream gcdumpStream = new(egressPath, FileMode.Open, FileAccess.Read);
+                    Assert.NotNull(gcdumpStream);
+
+                    ValidateGCDump(gcdumpStream);
+
+                    await runner.SendCommandAsync(TestAppScenarios.AsyncWait.Commands.Continue);
+                });
+            });
+        }
+
+        private static async void ValidateGCDump(Stream gcdumpStream)
```

Comment:
Do not use `async void` without a really good reason to do so. This should be `async Task`.

Reply:
Oops, good catch, that was an artifact from when I was playing around with trying FastSerialization and removed the `async` and I forgot to change it back.